client/src/main/proto/spine/validation.proto

```diff
@@ -47,15 +49,33 @@ extend google.protobuf.MessageOptions {
 
 extend google.protobuf.FieldOptions {
     RequiredOption required = 51002;
-    DecimalOption decimal = 51003;
-    DigitsOption digits = 51004;
-    TimeOption when = 51005;
+    DecimalMaxOption decimal_max = 51026;
+    DecimalMinOption decimal_min = 51037;
     MaxOption max = 51006;
     MinOption min = 51007;
+    DigitsOption digits = 51004;
+    TimeOption when = 51005;
     PatternOption pattern = 51008;
     ValidOption valid = 51009;
 }
 
+// A validation constraint violation for a message field.
+message ConstraintViolation {
+    // An interpolated error message for this constraint violation.
```

Comment:
Just copy-pasted this from Java Bean Validation docs.
In this context:
Interpolation - (in philology) modification of the original text; an insertion of words or phrases that did not appear in the original text by an interpreter.
I'll use a simpler term, though.